control/server.py

```diff
@@ -41,6 +42,17 @@ def callable():
 
     return callable
 
+def wait_for_file(file_path, timeout=None, interval=1):
+    start_time = time.time()
+
+    while True:
+        if os.path.exists(file_path):
+            return
+
+        if timeout is not None and time.time() - start_time >= timeout:
+            raise Exception(f"Timeout: file '{file_path}' not found after {timeout} seconds.")
```

Comment:
I know this is an [OS-related Exception](https://docs.python.org/3/library/exceptions.html#TimeoutError), but I find that non-generic `Exception`s are always friendlier to debug:
```suggestion
            raise TimeoutError(f"After {elapsed_time} seconds: {error_message}")
```